src/tfr_mission_control/src/tfr_mission_control/mission_control.cpp

```diff
@@ -0,0 +1,409 @@
+#include "tfr_mission_control/mission_control.h"
+
+#include <pluginlib/class_list_macros.h>
+
+namespace tfr_mission_control {
+
+    /* ========================================================================== */
+    /* Constructor/Destructor                                                     */
+    /* ========================================================================== */
+
+    /*
+     * First thing to get called, NOTE ros::init called by superclass
+     * Not needed here
+     * */
+    MissionControl::MissionControl()
+        : rqt_gui_cpp::Plugin(),
+        widget(nullptr),
+        autonomy{"autonomous_action_server",true},
+        teleop{"teleop_action_server",true},
+        com{nh.subscribe("com", 5, &MissionControl::updateStatus, this)},
+        teleopEnabled{false},
+        lastKey{0}
+    {
+        setObjectName("MissionControl");
+    }
+
+    /* NOTE: I think these QObjects have reference counting and are held by
+     * smart pointers under the hood. (I never ever see any qt programmers
+     * implement destructors or free memory on QObjects in any tutorial, example
+     * code or forum). However I do not know for sure, so I do it here to make 
+     * myself feel better, and it doesn't seem to complain.
+     * */
+    MissionControl::~MissionControl()
+    {
+        delete countdown;
+        delete motorKill;
+        delete widget;
+    }
+
+    /* ========================================================================== */
+    /* Initialize/Shutdown                                                        */
+    /* ========================================================================== */
+
+    /*
+     * Actually sets up all of the components on the screen and registers
+     * application context. Some of this code is boilerplate generated by that
+     * "catkin_creat_rqt" script, and it looked complicated, so I just took it
+     * at face value. I'll mark those sections with a //boilerplate annotation.
+     * */
+    void MissionControl::initPlugin(qt_gui_cpp::PluginContext& context)
+    {
+        //boilerplate
+        widget = new QWidget();
+        ui.setupUi(widget);
+        if (context.serialNumber() > 1)
+        {
+            widget->setWindowTitle(widget->windowTitle() +
+                    " (" + QString::number(context.serialNumber()) + ")");
+        }
+
+        //sets our window active, and makes sure we handle keyboard events
+        widget->setFocusPolicy(Qt::StrongFocus);
```

Comment:
I'm assuming that "Strong Focus" means that keyboard events outside the window don't trigger events here, correct?

Reply:
If that's the case that's a really good thing

Reply:
I also don't know I just followed the tutorial to get it to turn on, not a front end dev.